app/src/main/shared/actions/authActionCreators.ts

```diff
@@ -33,25 +34,39 @@ export const authActionCreators = {
         }
     },
 
-    loadSavedToken() {
-        return async (dispatch: Dispatch<any>) => {
-            const token = localStorageHandler.get("accessToken");
-            if (token) {
-                const inflated = jwtTokenAuth.inflateToken(token);
-                const decoded = inflated && jwtTokenAuth.decodeToken(inflated);
+    loadAuthenticatedUser() {
+        // fetch details of currently logged in user from API.
+        return async (dispatch: Dispatch<any>, getState: () => GlobalState) => {
+            try {
+                const userResponse = await(new AuthService(dispatch, getState)).getCurrentUser();
+                //TODO: since we should always have the modelling groups for the current user, either from the API here
```

Comment:
Does this sound right to you? I remember adding this endpoint to get modelling groups for the current user, but now not sure why we needed it if the modelling groups were already embedded in the token! Arguably we could roll this endpoint into the currentUser one if this is the only place it's needed. Shall I create another ticket for that?